config.go

```diff
@@ -0,0 +1,224 @@
+// Copyright 2022 Juan Pablo Tosso and the OWASP Coraza contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package coraza
+
+import (
+	"io/fs"
+
+	"github.com/corazawaf/coraza/v3/internal/corazawaf"
+	"github.com/corazawaf/coraza/v3/loggers"
+	"github.com/corazawaf/coraza/v3/types"
+)
+
+type WAFConfig interface {
+	WithRule(rule *corazawaf.Rule) WAFConfig
```

Comment:
How about accepting a variadic set of rules? This will save `clone` calls and will allow a more fluent API.

Reply:
Probably also renaming it to `WithRules`.

Reply:
Shall we take it post merge? @anuraaga

Reply:
Oh sorry yeah forgot about this one will follow up